orm/internal/codegen/file.go

```diff
@@ -0,0 +1,118 @@
+package codegen
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/cosmos/cosmos-proto/generator"
+	"github.com/iancoleman/strcase"
+	"google.golang.org/protobuf/compiler/protogen"
+	"google.golang.org/protobuf/proto"
+
+	v1alpha1 "github.com/cosmos/cosmos-sdk/api/cosmos/orm/v1alpha1"
+)
+
+var (
+	tablePkg = protogen.GoImportPath("github.com/cosmos/cosmos-sdk/orm/model/ormtable")
+)
+
+type fileGen struct {
+	*generator.GeneratedFile
+	file *protogen.File
+}
+
+func (f fileGen) gen() error {
+	f.P("// Code generated by protoc-gen-go-cosmos-orm. DO NOT EDIT.")
+	f.P()
+	f.P("package ", f.file.GoPackageName)
+	f.genStoreInterface()
+	for _, msg := range f.file.Messages {
+		tableDesc := proto.GetExtension(msg.Desc.Options(), v1alpha1.E_Table).(*v1alpha1.TableDescriptor)
```

Comment:
we need to check for singletons too and their interface should probably be very simple:

```go
type ExampleSingletonStore interface {
  Get() (*ExampleSingleton, error)
  Save(*ExampleSingleton) error
}
```